UI/GamepadEmu.h

```diff
@@ -289,25 +313,44 @@ namespace CustomKey {
 	static_assert(ARRAY_SIZE(comboKeyList) <= 64, "Too many key for a uint64_t bit mask");
 };
 
-class GestureGamepad : public UI::View {
-public:
-	GestureGamepad() {};
-
-	void Touch(const TouchInput &input) override;
-	void Update() override;
-
-protected:
-
-	float lastX_ = 0.0f;
-	float lastY_ = 0.0f;
-	float deltaX_ = 0.0f;
-	float deltaY_ = 0.0f;
-	float lastTapRelease_ = 0.0f;
-	float lastTouchDown_ = 0.0f;
-	int dragPointerId_ = -1;
-	bool swipeLeftReleased_ = true;
-	bool swipeRightReleased_ = true;
-	bool swipeUpReleased_ = true;
-	bool swipeDownReleased_ = true;
-	bool haveDoubleTapped_ = false;
-};
+// Gesture key only have virtual button that can work without constant press
+namespace GestureKey {
+	struct key {
+		const char* n; // UI name
+		uint32_t c; // Key code
+	};
+	static const key keyList[] = {
+		{ "Square", CTRL_SQUARE },
+		{ "Triangle", CTRL_TRIANGLE },
+		{ "Circle", CTRL_CIRCLE },
+		{ "Cross", CTRL_CROSS },
+		{ "Up", CTRL_UP },
```

Comment:
Hm, not sure exactly how, but wish we could reuse the strings that exist elsewhere for the same purpose in the main control mapper.

Reply:
We had a method already to get the name in a `std::string`, tho' I added  a new one to get the `char *` as the string get translated when you click on the choice and the c_str() pointer might be invalid by then.

Made the same for combo button while was at it :)